### PR TITLE
Heatmap with weighted points and map styles

### PR DESCRIPTION
**Pull request by nziehn**
heatmap_weighted(lats, lngs, weights, ) allows creating heatmaps with weigthed points, 

gmap.draw("mymap.html", map_styles=[...]) can be used to style the maps according to https://developers.google.com/maps/documentation/javascript/styling

### CODE REVIEW

File: gmplot/gmplot.py [hunk 1]
```diff
@@ -269,12 +269,16 @@ def write_shapes(self, f):
             self.write_polygon(f, shape, settings)
 
     # TODO: Add support for mapTypeId: google.maps.MapTypeId.SATELLITE
-    def write_map(self,  f):
+    def write_map(self,  f, map_styles=None):
```
Comment (vgm64) [inline gmplot/gmplot.py]: Instead of map_styles=None, how about map_styles=[]?

Comment (nziehn) [inline gmplot/gmplot.py]: We should avoid mutable objects in default function parameters:
http://docs.python-guide.org/en/latest/writing/gotchas/

**Comment by frslm** [comment]
Hey @nziehn I know it's been quite a while since this was last looked at, but would you be able to split the map styles portion into a separate PR? I'd like to review and merge the map styles enhancement separately.

**Comment by frslm** [comment]
I've rebased and merged the map styles portion of this PR in #148 since I don't think I can modify this PR.

As for the weighted heatmap portion, that's been implemented in #132 (I had submitted that PR before I realized this PR tackled weighted heatmaps too, sorry about that).

Since all the features in this PR have been implemented, I'll be closing this.

**Pull request closed without merging**